broccoli/version.js

```diff
@@ -25,8 +25,12 @@ module.exports.VERSION = (() => {
   // * current "build type" or branch name (in CI this is generally not
   //   present, but it is very useful for local / testing builds)
   // * the sha for the commit
-  let packageVersion  = pkg.version;
+  let packageVersion = pkg.version;
   let sha = info.sha || '';
+  // * remove illegal non-alphanumeric characters from branch name.
+  if(info.branch !== null && new RegExp(/[^a-zA-Z\d\s-]/).test(info.branch)) {
+    info.branch = info.branch.replace(/[^a-zA-Z\d\s-]/, '-');
```

Comment:
Would you mind tweaking this a tad bit, I'd like to avoid mutating the `info` object here and avoid repeating the regexp.

I _think_ this might do the trick:

```js
// ...snip...
let sha = info.sha || '';
let suffix = process.env.BUILD_TYPE || info.branch;
suffix = suffix.replace(/[^a-zA-Z\d\s-]/, '-');
// ...snip...
```

@Parrryy - What do you think?

Reply:
Sounds good to me.

Reply:
@rwjblue What about the case when the suffix is `process.env.BUILD_TYPE`? I'm not exactly sure what that is and if the `.replace` will effect it.